### ISSUE TITLE
Android Build Error

### ISSUE THREAD

**Issue by tomfebry** [getsentry/sentry-cordova#159]
I am using ionic with Cordova and when I attempt to build an Android APK via the Ionic build server, I get the following build error (snippet):

```
[12:35:34]: ▸ Exception in thread "main" java.lang.IllegalArgumentException
[12:35:34]: ▸ at org.objectweb.asm.ClassReader.(Unknown Source)
[12:35:34]: ▸ at org.objectweb.asm.ClassReader.(Unknown Source)
[12:35:34]: ▸ at org.objectweb.asm.ClassReader.(Unknown Source)
[12:35:34]: ▸ at com.google.devtools.build.android.desugar.CoreLibraryRewriter.reader(CoreLibraryRewriter.java:44)
[12:35:34]: ▸ at com.google.devtools.build.android.desugar.Desugar.desugarClassesInInput(Desugar.java:388)
[12:35:34]: ▸ at com.google.devtools.build.android.desugar.Desugar.desugarOneInput(Desugar.java:326)
[12:35:34]: ▸ at com.google.devtools.build.android.desugar.Desugar.desugar(Desugar.java:280)
[12:35:34]: ▸ at com.google.devtools.build.android.desugar.Desugar.main(Desugar.java:584)
[12:35:34]: ▸ FAILURE: Build failed with an exception.
[12:35:34]: ▸ * What went wrong:
[12:35:34]: ▸ Execution failed for task ':app:transformClassesWithDesugarForDebug'.
[12:35:34]: ▸ > com.android.build.api.transform.TransformException: java.lang.RuntimeException: com.android.ide.common.process.ProcessException: Error while executing java process with main class com.google.devtools.build.android.desugar.Desugar
```

iOS seems to compile fine, is anyone else having this problem or know a way around this?

I have verified that removing the Cordova plugin allows the app to build, but when adding it back in, it causes this error.

**Comment by thaitrungnguyen** [comment]
I can also verify the problem. Trying to fix the problem since days.

**Comment by tomfebry** [comment]
I ended up now using this plugin and just using pure JS, AFAIK this is just for catching native crash errors, the JS library catches the errors produced in the web layer (using ionic-angular)

**Comment by Kiludude** [comment]
Got the same issue and I'm actually trying to pick up native crash errors. The options for this seem to be somewhat limited. Having a look at Crashlytics now.

**Comment by sanjredfield** [comment]
Got the same issue when upgrading, v0.15.0 seems to be working fine.

**Comment by Kiludude** [comment]
> Got the same issue when upgrading, v0.15.0 seems to be working fine.

Really? Just tried installing v0.15.0 but I get the exact same error.

**Comment by sanjredfield** [comment]
> > Got the same issue when upgrading, v0.15.0 seems to be working fine.
> 
> Really? Just tried installing v0.15.0 but I get the exact same error.

My bad. Build wasn't actually working - I thought the plugin was added again but it wasn't. Sorry for wasting your time!

**Comment by mzoeller** [comment]
I can also confirm this.

**Comment by renanbandeira** [comment]
Having same issue. Using cordova 7.1.0, android 7.0. I'm trying to move from Fabric Crashlytics to sentry-cordova since I already use Sentry for JS layer.

**Comment by Prateek-Kickdrum** [comment]
Facing Same Issue. 
Any update on this? 

Using Cordova 8.1.2
Android 7.1.1
Sentry-cordova 0.17.0

**Comment by Prateek-Kickdrum** [comment]
I think I am able to build the android app now.

I just update the sentry-android version in build.gradle file in the project.

Before: 
`io.sentry:sentry-android:1.+`

Output: Giving above error.

After:
`io.sentry:sentry-android:1.7.27`

Output: Build is successful.

Hope it helps.

**Comment by mzoeller** [comment]
Thanks! I can confirm this finally fixed the problem for me, too. I had to edit _platforms/android/project.properties_:
`cordova.system.library.7=io.sentry:sentry-android:1.7.27+`

**Comment by renanbandeira** [comment]
It actually fixes the problem! Thanks for the help. Now my config.xml looks like:
```
<plugin name="sentry-cordova" spec="~0.17.0">
        <variable name="SENTRY_ANDROID_SDK_VERSION" value="1.7.27"/>
 </plugin>
```

And works perfectly

**Comment by neogenz** [comment]
Thanks guys that works perfectly for me ! I've used the solution of @renanbandeira to don't have to re-edit build.gradle file at each platform rm/add.

My config :

Ionic:

   Ionic CLI          : 5.2.8 
   Ionic Framework    : ionic-angular 3.9.2
   @ionic/app-scripts : 3.1.4

Cordova:

   Cordova CLI       : 8.1.2 (cordova-lib@8.1.1)
   Cordova Platforms : android 7.1.4
   Cordova Plugins   : cordova-plugin-ionic-webview 1.2.1, (and 7 other plugins)

Utility:

   cordova-res : 0.6.0 (update available: 0.11.0)
   native-run  : 0.2.8 (update available: 0.3.0)

System:

   Android SDK Tools : 26.1.1
   ios-deploy        : 1.9.4
   NodeJS            : v10.16.0
   npm               : 6.9.0
   OS                : macOS Catalina
   Xcode             : Xcode 11.4 Build version 11E146

**Comment by jennmueng** [comment]
We've bumped the sentry-android SDK up in our latest version [1.0.0-rc.0](https://github.com/getsentry/sentry-cordova/releases/tag/v1.0.0-rc.0) so I'll be closing this along with the fact it hasn't seen updates in a while. Please let us know if it occurs again!
